bitnami/kafka/templates/svc-external-access.yaml

```diff
@@ -0,0 +1,44 @@
+{{- if .Values.externalAccess.enabled -}}
+  {{- $fullName := include "kafka.fullname" . }}
+  {{- $replicaCount := .Values.replicaCount | int }}
+  {{- $root := . }}
+
+{{- range $i, $e := until $replicaCount }}
+  {{- $targetPod := printf "%s-%d" (printf "%s" $fullName) $i }}
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "kafka.fullname" $ }}-{{ $i }}-external
+  labels: {{- include "kafka.labels" $ | nindent 4 }}
+    app.kubernetes.io/component: kafka
+    pod: {{ $targetPod }}
+spec:
+  type: {{ $root.Values.externalAccess.service.type }}
+  {{- if eq $root.Values.externalAccess.service.type "LoadBalancer" }}
+  {{- if $root.Values.externalAccess.service.loadBalancerIP }}
+  loadBalancerIP: {{ index $root.Values.externalAccess.service.loadBalancerIP $i }}
+  {{- end }}
+  {{- end }}
+  ports:
+    - name: kafka
+      {{- if eq $root.Values.externalAccess.service.type "LoadBalancer" }}
+      port: {{ $root.Values.externalAccess.service.port }}
+      {{- else }}
+      port: {{ index $root.Values.externalAccess.service.nodePort $i }}
+      {{- end }}
+      {{- if and $root.Values.externalAccess.service.nodePort (eq $root.Values.externalAccess.service.type "NodePort") }}
+      nodePort: {{ index $root.Values.externalAccess.service.nodePort $i }}
+      {{- end }}
```

Comment:
A "LoadBalancer" service also exposes a node port, please remove the second condition

Reply:
done